src/Uri.php

```diff
@@ -1352,7 +1352,7 @@ private function filterString($str): ?string
     /**
      * {@inheritDoc}
      */
-    public function withUserInfo($user, $password = null): UriInterface
+    public function withUserInfo($user, ?string $password = null): UriInterface
```

Comment:
you can't add this typehinting. One of the reason I went for PHP7.2 was because it introduces [type widening which enables complying to an interface while being able to accept more arguments type that expected by the interface](https://wiki.php.net/rfc/parameter-no-type-variance).